spec/models/project_spec.rb

```diff
@@ -1937,25 +1937,50 @@
   end
 
   describe "#color" do
-    before  { CatarseSettings[:default_color] = '#ff8a41' }
-    subject { project.color }
+    before { CatarseSettings[:default_color] = '#ff8a41' }
 
-    context "when a project has a recurring channel" do
-      let(:default_color) { CatarseSettings[:default_color] }
-      let(:channel)       { create :channel, recurring: true }
-      let(:project)       { create :project, channels: [channel] }
+    context "when the project is in a channel" do
+      context "and the channel's main color has value" do
+        let(:project)    { create(:project, channels: [channel]) }
+        let(:channel)    { create(:channel, main_color: main_color) }
+        let(:main_color) { 'main' }
+
+        it "returns the channel's main color" do
+          expect(project.color).to eq main_color
+        end
+      end
+
+      context "and the channel's main color is empty" do
+        context "when the project has category" do
+          let(:project)        { create(:project, channels: [channel], category: category) }
+          let(:channel)        { create(:channel, :non_recurring, main_color: '') }
+          let(:category)       { create(:category, color: category_color) }
+          let(:category_color) { 'c_color' }
 
-      it "should return the default color" do
-        is_expected.to eq(default_color)
+          it "returns the category's color" do
+            expect(project.color).to eq category_color
```

Comment:
Just a small detail.
I think you don't need `let(:category_color)...`, you can do it straight forward by just adding the color direct in the category, and in the test just call `category.color`. By doing this you, when you read the test, you enforce that the color comes from the `category`.
What do you think?
```ruby
....
let(:category) { create(:category, color: 'c_color' }

 it "returns the category's color" do
   expect(project.color).to eq category.color
.....
````


Same comment about the `main_color` in the `channel`